src/Confluent.Kafka/Consumer.cs

```diff
@@ -266,6 +266,38 @@ IAsyncDeserializer<TValue> taskValueDeserializer
             => (keyDeserializer != null && valueDeserializer != null)
                 ? Consume<TKey, TValue>(timeout.TotalMillisecondsAsInt(), keyDeserializer, valueDeserializer) // fast path for simple case
                 : Consume(timeout.TotalMillisecondsAsInt());
+
+
+        /// <summary>
+        ///     Commits an offset based on the topic/partition/offset of a ConsumeResult.
+        /// </summary>
+        /// <param name="result">
+        ///     The ConsumeResult instance used to determine the committed offset.
+        /// </param>
+        /// <param name="cancellationToken">
+        ///     A cancellation token that can be used to cancel this operation
+        ///     (currently ignored).
+        /// </param>
+        /// <exception cref="Confluent.Kafka.KafkaException">
+        ///     Thrown if the request failed.
+        /// </exception>
+        /// <exception cref="Confluent.Kafka.TopicPartitionOffsetException">
+        ///     Thrown if the result is in error.
+        /// </exception>
+        /// <remarks>
+        ///     A consumer which has position N has consumed messages with offsets up to N-1 
+        ///     and will next receive the message with offset N. Hence, this method commits an 
+        ///     offset of <paramref name="result" />.Offset + 1.
+        /// </remarks>
+        public void Commit(ConsumeResult<TKey, TValue> result, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (result.Message == null)
+            {
+                throw new InvalidOperationException("Attempt was made to commit offset corresponding to an empty consume result");
+            }
+
+            Commit(new [] { new TopicPartitionOffset(result.TopicPartition, result.Offset + 1) });
```

Comment:
Is an exception raised on commit error?

Reply:
yes

Reply:
for your reference: https://github.com/confluentinc/confluent-kafka-dotnet/blob/master/src/Confluent.Kafka/Impl/SafeKafkaHandle.cs#L688

all the different error cases throw an appropriate exception. whilst we're on the topic though, we need to introduce more exception types. These will derive from `KafkaException` and `TopicPartitionOffsetException`. Since doing that is not a breaking change, I think an overhaul of exceptions can come after v1.0.